soroban-env-host/src/host/frame.rs

```diff
@@ -736,39 +736,17 @@ impl Host {
     // Make the in-memory instance storage persist into the `Storage` by writing
     // its updated contents into corresponding `ContractData` ledger entry.
     fn persist_instance_storage(&self) -> Result<(), HostError> {
-        let updated_instance = self.with_current_context_mut(|ctx| {
+        let updated_instance_storage = self.with_current_context_mut(|ctx| {
             if let Some(storage) = &ctx.storage {
                 if !storage.is_modified {
                     return Ok(None);
                 }
-                let executable = match &ctx.frame {
-                    Frame::ContractVM { instance, .. } => {
-                        instance.executable.metered_clone(self)?
-                    }
-                    Frame::HostFunction(_) => {
-                        return Err(self.err(
-                            ScErrorType::Context,
-                            ScErrorCode::InternalError,
-                            "unexpected storage for host fn",
-                            &[],
-                        ))
-                    }
-                    Frame::Token(_, _, _, instance) => instance.executable.metered_clone(self)?,
-                    // Mock executable for test contract 'instances'. This is
-                    // just a placeholder - it's not used for actually calling
-                    // the test contracts.
-                    #[cfg(any(test, feature = "testutils"))]
-                    Frame::TestContract(t) => t.instance.executable.clone(),
```

Comment:
Where are the token and test contract cases handled now? I don't see where.

Reply:
We don't need to handle these cases at all because we don't care what's the *current* executable. We just need to preserve the executable in the instance entry, which we do by passing `None` into `store_contract_instance`.